xla/service/cpu/onednn_matmul.cc

```diff
@@ -302,6 +302,30 @@ ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnMatMul(
   TRANSPOSE_LAST_TWO_DIMS_IF(
       matmul_config.transpose_b() && weights_md.get_ndims() > 1, weights_md);
   auto output_md = output_minfo.GetOneDnnMemDesc();
+
+  // TODO(snadampal): allocate the reordered weights Literal conditionally
+  // only when the weights are already not pre-packed and only for aarch64
+  // currently it's allocated upfront because Literal object scope issues
+  auto bias_md = dnnl::memory::desc{};
+  if (absl::c_count(matmul_config.fusions().ops(), OneDnnFusionConfig::BIAS) >
+        0) {
+      MemrefInfo bias_minfo(args[arg_indx++]);
```

Comment:
Hi @akhilgoe , this is conditionally done only is there was a bias operand. so, the args array would have all the required indices.
to make sure i understand your point, could you please elaborate it?

Reply:
Sure, what I mean to say is that this `if-block` is only being used to get the bias md, but it has a side-effect, it is incrementing `arg_indx`, iow `arg_indx` now points to the argument after bias, and since you are not storing the pod data anywhere my concern is that the bias data is lost.

Reply:
I see your point now, thank you! I have updated the PR. Since the reorder is an additional primitive i have introduced, I'm just skipping the  index increment, so that the actual matmul primitive processing is not affected.

Reply:
Hi @akhilgoe, please let me know if this addresses your comment.

Reply:
Yes it does, thanks! 
Can you please run a model that will trigger the MM + Bias ( + activations) scenario and / or run tests under xla/service/cpu/tests/onednn_matmul_test.cc and confirm that everything runs as expected, thanks!

Reply:
Hi @akhilgoe , I have updated the PR and with this version all the unit tests are passing on AWS Graviton4 based c8g instances. appreciate if you could review one more time.

note: the one test failure from `onednn_matmul_test` ([  FAILED  ] MatmulTest.TestF32ConstantWeights (77 ms))
 is not related to this PR, it's failing on the mainline code as well, so, planning to take that up separately instead of blocking this PR.

~~~
//xla/service/cpu/tests:cpu_bytesizeof_test                     (cached) PASSED in 0.2s
//xla/service/cpu/tests:onednn_matmul_test                  (3/4 cached) FAILED in 1 out of 4 in 13.1s
  Stats over 4 runs: max = 13.1s, min = 2.0s, avg = 5.5s, dev = 4.5s
  /home/ubuntu/.cache/bazel/_bazel_ubuntu/6d062a7b894a0860cec805fab2fe3ab8/execroot/xla/bazel-out/aarch64-opt/testlogs/xla/service/cpu/tests/onednn_matmul_test/shard_1_of_4/test.log
//xla/service/cpu/tests:cpu_aot_export_test                              PASSED in 1.4s
//xla/service/cpu/tests:cpu_dyn_shape_test                               PASSED in 1.4s
//xla/service/cpu/tests:cpu_eigen_dot_operation_test                     PASSED in 1.3s
//xla/service/cpu/tests:cpu_external_constants_test                      PASSED in 1.3s
//xla/service/cpu/tests:cpu_fusion_test                                  PASSED in 1.2s
//xla/service/cpu/tests:cpu_infeed_test                                  PASSED in 1.2s
//xla/service/cpu/tests:cpu_intrinsic_test                               PASSED in 1.5s
//xla/service/cpu/tests:cpu_key_value_sort_test                          PASSED in 1.4s
//xla/service/cpu/tests:cpu_literal_caching_test                         PASSED in 1.4s
//xla/service/cpu/tests:cpu_noalias_test                                 PASSED in 1.2s
//xla/service/cpu/tests:cpu_outfeed_test                                 PASSED in 1.4s
//xla/service/cpu/tests:cpu_profiling_test                               PASSED in 1.4s
//xla/service/cpu/tests:cpu_spmd_compile_test                            PASSED in 1.4s
//xla/service/cpu/tests:cpu_topk_test                                    PASSED in 1.4s
//xla/service/cpu/tests:cpu_vectorization_test                           PASSED in 1.4s
//xla/service/cpu/tests:cpu_while_test                                   PASSED in 1.4s
//xla/service/cpu/tests:onednn_convolution_test                          PASSED in 1.4s
//xla/service/cpu/tests:onednn_layer_norm_test                           PASSED in 3.0s
//xla/service/cpu/tests:tree_reduction_rewriter_test                     PASSED in 1.4s
//xla/service/cpu/tests:onednn_softmax_test                              PASSED in 3.2s
  Stats over 4 runs: max = 3.2s, min = 1.3s, avg = 1.8s, dev = 0.8s

Executed 21 out of 22 tests: 21 tests pass and 1 fails locally.
~~~